src/audio_excl.rs

```diff
@@ -16,70 +15,66 @@ pub struct FileLister {
             .map(|s| s.path())
-            .choose_multiple(&mut rng, num_items);
-        println!("Got items in order {:?}", items);
-        return Ok(FileLister {
-            files: items,
-            lock_acq: HashMap::new(),
-            security: HashMap::new(),
-            timeout: *timeout,
-        });
+            .collect::<Vec<_>>();
+        files.shuffle(&mut rand::thread_rng());
+        println!("Got items in order {:?}", files);
+        Ok(FileLister {
+            files,
+            timeout,
+            ..Default::default()
+        })
     }
 
-    pub fn clean(&mut self) -> io::Result<()> {
-        let mut add_back: Vec<PathBuf> = vec![];
-        let timeout = self.timeout.clone();
+    pub fn clean(&mut self) {
+        let mut add_back = Vec::new();
+        let timeout = &self.timeout;
         self.lock_acq.retain(|path, modified| {
-            let duration = match SystemTime::now().duration_since(*modified) {
-                Err(_) => return false,
-                Ok(t) => t,
-            };
-            if duration > timeout {
-                add_back.push(path.to_path_buf());
-                false
-            } else {
-                true
-            }
+            SystemTime::now()
+                .duration_since(*modified)
+                .map(|duration| {
+                    if duration > *timeout {
+                        add_back.push(path.to_path_buf());
+                        false
+                    } else {
+                        true
+                    }
+                })
+                .unwrap_or(false)
         });
         self.files.append(&mut add_back);
-        Ok(())
     }
 
     pub fn get_file(&mut self) -> io::Result<(String, String)> {
         // if opening item yields an error, just ditch it
-        let filename = match self.files.pop() {
-            Some(path) => path,
-            None => return Err(io::Error::new(io::ErrorKind::NotFound, "No more files")),
-        };
+        let filename = self
+            .files
+            .pop()
+            .ok_or_else(|| io::Error::new(io::ErrorKind::NotFound, "No more files"))?;
         self.lock_acq
             .insert(filename.to_path_buf(), SystemTime::now());
         let new_uuid = Uuid::new_v4();
         self.security
             .insert(filename.to_path_buf(), new_uuid.to_string());
-        let name_w_extension = match filename.file_name() {
-            None => {
-                return Err(io::Error::new(
+        filename
+            .file_name()
+            .ok_or_else(|| {
+                io::Error::new(
                     io::ErrorKind::InvalidData,
                     "found a filename ending with ..",
-                ))
-            }
-            Some(p) => match p.to_str() {
-                None => {
-                    return Err(io::Error::new(
-                        io::ErrorKind::InvalidData,
-                        "Somehow this was invalid?",
```

Comment:
`PathBuf::to_str` only fails when the underlying `OsString` doesn't contain valid UTF-8 (Windows allows [lone surrogates](https://unicodebook.readthedocs.io/unicode_encodings.html#utf-16-surrogate-pairs), for example, and on Linux paths are pretty much unrestricted byte-arrays).